src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -140,7 +142,9 @@ internal static SourcePropertySymbol Create(SourceMemberContainerTypeSymbol cont
             {
                 Binder.CheckFeatureAvailability(
                     syntax,
-                    (hasGetAccessor && !hasSetAccessor) ? MessageID.IDS_FeatureReadonlyAutoImplementedProperties : MessageID.IDS_FeatureAutoImplementedProperties,
+                    hasGetAccessor && hasSetAccessor ?
+                        (hasAutoPropertyGet && hasAutoPropertySet ? MessageID.IDS_FeatureAutoImplementedProperties : MessageID.IDS_FeatureFieldKeyword) :
+                        MessageID.IDS_FeatureReadonlyAutoImplementedProperties,
```

Comment:
nit: It seems like the new factoring is going to diagnose a setter-only auto property (error case) as requiring the readonly auto-properties feature.

Reply:
Good catch, thanks.